From fcd3b2f8b72509001876b0253f0ad1013704fe08 Mon Sep 17 00:00:00 2001
Message-ID: <fcd3b2f8b72509001876b0253f0ad1013704fe08.1732837202.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1732837202.git.sharlatanus@gmail.com>
References: <cover.1732837202.git.sharlatanus@gmail.com>
From: "Artyom V. Poptsov" <poptsov.artyom@gmail.com>
Date: Sun, 24 Nov 2024 13:27:34 +0300
Subject: [PATCH 08/15] gnu: Add go-github-com-charmbracelet-x-term.

* gnu/packages/golang-xyz.scm (go-github-com-charmbracelet-x-term): New variable.

Change-Id: I854febc3a3010f39051ee6fbd4a159993043d41e
Reviewed-by: Sharlatan Hellseher <sharlatanus@gmail.com>
---
 gnu/packages/golang-xyz.scm | 33 +++++++++++++++++++++++++++++++++
 1 file changed, 33 insertions(+)

diff --git a/gnu/packages/golang-xyz.scm b/gnu/packages/golang-xyz.scm
index 81fd07f402..81dc15ae5d 100644
--- a/gnu/packages/golang-xyz.scm
+++ b/gnu/packages/golang-xyz.scm
@@ -1589,6 +1589,39 @@ (define-public go-github-com-charmbracelet-x-ansi
 ECMA-48} specs.")
     (license license:expat)))
 
+(define-public go-github-com-charmbracelet-x-term
+  (package
+    (name "go-github-com-charmbracelet-x-term")
+    (version "0.2.1")
+    (source
+     (origin
+       (method git-fetch)
+       (uri (git-reference
+             (url "https://github.com/charmbracelet/x")
+             (commit (go-version->git-ref version
+                                          #:subdir "term"))))
+       (file-name (git-file-name name version))
+       (sha256
+        (base32 "1shw55110fnn4xz80wmgr18czmiil6z1j064m90iw8c7j9llfzn5"))))
+    (build-system go-build-system)
+    (arguments
+     (list
+      #:import-path "github.com/charmbracelet/x/term"
+      #:unpack-path "github.com/charmbracelet/x"))
+    (propagated-inputs (list go-github-com-rivo-uniseg
+                             go-golang-org-x-sys))
+    (home-page "https://github.com/charmbracelet/x")
+    (synopsis "Terminal utilities and helpers")
+    (description
+     "@code{term} provides an API for working with terminals that includes:
+@itemize
+@item Switching a terminal to the raw mode.
+@item Getting, setting and restoring the state of a terminal.
+@item Getting size of a terminal.
+@item Reading passwords from a terminal without a local echo.
+@end itemize")
+    (license license:expat)))
+
 (define-public go-github-com-charmbracelet-x-exp-golden
   (package
     (name "go-github-com-charmbracelet-x-exp-golden")
-- 
2.46.0

